analyzers/src/SonarAnalyzer.CSharp/Rules/SetPropertiesInsteadOfMethods.cs

```diff
@@ -0,0 +1,33 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class SetPropertiesInsteadOfMethods : SetPropertiesInsteadOfMethodsBase<SyntaxKind, InvocationExpressionSyntax>
+{
+    protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
+
+    protected override bool HasCorrectArgumentCount(InvocationExpressionSyntax invocation) =>
```

Comment:
I've implemented a `HasExactlyNArguments` in the facade in [this PR](https://github.com/SonarSource/sonar-dotnet/pull/7208/files#diff-16a2d259d04a1b16e0c004f5d441adaf8e932260e8a07facf9e70e9d4bd123fc). You can copy the implementation and use it here too, this way the language-specific files should be empty

Reply:
Even if I have it in the facade, the count is different:

```
set.Min(); // C#, 0 arguments
Enumerable.Min(set); // VB.NET, 1 argument
```

So I don't think I have a huge benefit for de-duplication here.

Reply:
That's unfortunate, resolving the conversation 👍